app/controllers/posts_controller.rb

```diff
@@ -1,7 +1,15 @@
 class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
-    @posts = @user.posts.includes(:comments)
+    if current_user.nil?
+      redirect_to new_user_session_path
+      return
+    end
+    @posts = if current_user.role?
+               @user.posts.includes(:comments)
+             else
+               @user.posts.includes(:comments, :author)
```

Comment:
Kindly consider adding `load_and_authorize_resource` in the Post Controller in order to automatically set the authorizations for the controller. By calling `load_and_authorize_resource` on a controller, most of the CRUD method calls on models are added automatically. :wink:

Reply:
Hello @jerryowusu!!!

Thanks for the review! I would like to ask if this is a project requirement? I didn't see it anywhere and i managed to do it manually. Changing everything and adding `load_and_authorize_resource` will delay my progress. Is the app working now on your side or not please?